### ISSUE TITLE
patch: allow different "ignore_discard" value at save() time [rt.cpan.org #85011]

### ISSUE THREAD

**Issue by oalders** [libwww-perl/HTTP-Cookies#23]
Migrated from [rt.cpan.org#85011](https://rt.cpan.org/Ticket/Display.html?id=85011) (status was 'new')

Requestors:
* lxp@cpan.org

From lxp@cpan.org on 2013-05-03 05:45:55:
```
In HTTP::Cookies v6.01, the "save" method only saves "Discard" cookies
if the "ignore_discard" attribute was set to a true value at object
construction.

If that did not occur, then currently there is no way to cause "Discard"
cookies to be written to file without breaking object encapsulation.

This is a problem when code wishing to save the complete contents of a
cookie jar does not have control over its creation.

Please consider adding either:

1.  an "ignore_discard" method to allow altering of this attribute after
    construction; or

2.  a second, optional argument to the "save" method which allows
    overriding of the object-based "ignore_discard" attribute:

    $cookie_jar->save( $file, $ignore_discard )
```

From lxp@cpan.org on 2013-05-03 05:54:33:
```
On Fri May 03 15:45:55 2013, LXP wrote:
> Please consider adding either:
> 
> 1.  an "ignore_discard" method to allow altering of this attribute after
>     construction; or
> 
> 2.  a second, optional argument to the "save" method which allows
>     overriding of the object-based "ignore_discard" attribute:
> 
>     $cookie_jar->save( $file, $ignore_discard )

After further thought, I think I would personally prefer option #2 since
it doesn't alter object state (and therefore won't cause unintended side
effects no possible later calls to "save").

Both options are backwards-compatible with the current API, however.
```

From lxp@cpan.org on 2013-05-03 05:55:22:
```
On Fri May 03 15:54:33 2013, LXP wrote:
> After further thought, I think I would personally prefer option #2 since
> it doesn't alter object state (and therefore won't cause unintended side
> effects no possible later calls to "save").

...effects on* possible later calls to "save."
```

From lxp@cpan.org on 2013-05-03 06:32:07:
```
A test and patch for this functionality (implemented as an additional
save() argument) is available here:

https://github.com/gisle/http-cookies/pull/2
```

**Comment by Lx** [comment]
As this issue nears its six birthday, is there anything that significantly impedes its resolution?

**Comment by oalders** [comment]
Sorry @Lx there's a real shortage of people power here.  Would you be able to fix and force-push the merge conflict in https://github.com/libwww-perl/http-cookies/pull/2 and then we can discuss it?

**Comment by oalders** [comment]
Also, it would be helpful to add an appropriate line to the Changes file.

**Comment by Lx** [comment]
I've addressed the merge conflict and added a line to the `Changes` file as requested.

**Comment by oalders** [comment]
Closed via #2
